pandas/plotting/_matplotlib/core.py

```diff
@@ -1188,6 +1189,8 @@ def _post_plot_logic(self, ax, data):
         from matplotlib.ticker import FixedLocator
 
         def get_label(i):
+            if is_float(i) and i.is_integer():
+                i = int(i)
```

Comment:
I am not really sure this is correct, but it's what is currently actually is being done (the conversion to int is just happening inside the `Index.__getitem__`). So this change is needed to preserve the current behaviour.

It's a bit strange that those xticks are floats, though. But maybe that's just a matplotlib behaviour.

Reply:
this is really strange ,what is i here?

Reply:
It are the values from `ax.get_xticks()`, see 5 lines below. I suppose matplotlib just always returns floats

Reply:
Yes, just checked, matplotlib simply always returns floats. And it's only for the "integer floats" that we have a value in the index to put as xtick label.